plugins/manipulator.cpp

```diff
@@ -1173,43 +2150,118 @@ viewscreen_unitlaborsst::viewscreen_unitlaborsst(vector<df::unit*> &src, int cur
 
         units.push_back(cur);
     }
-    altsort = ALTSORT_NAME;
-    detail_mode = DETAIL_MODE_PROFESSION;
-    first_column = sel_column = 0;
 
+    if(cur_world!=World::ReadWorldFolder()){
+        cur_world = World::ReadWorldFolder();
+        resetModes();
+    }
+
+    row_hint=0;
+    col_hint=0;
     refreshNames();
+    calcArrivals();
+
+    /// How to pass selected unit back and fro the df unit listing ?
+    /// When we go to view crea from here, df gets the selected unit
+    /// but when we just leave here, it does not ....
+    /*
+    df::unit *cpunit = findCPsActiveUnit(cursor_pos);
+
+    if(cpunit){
+        int sel_acx = active_index[cpunit];
+        for (size_t i = 0; i < units.size(); i++)
+        {
+            if(sel_acx==units[i]->active_index){
+                sel_row=i;
+                break;
+            }
+        }
+    }
+    */
+
+    if(sel_row>=units.size())
+        sel_row=units.size()-1;
+    int sel_row_a = sel_row;
+
     calcIDs();
+    unit_info_ops::calcAptScores(units);
+    unit_info_ops::calcNotices(units);
+
+    unstashSelection(units);
+    dualSort();
 
-    first_row = 0;
-    sel_row = cursor_pos;
-    calcSize();
+    sizeDisplay();
 
-    // recalculate first_row to roughly match the original layout
-    first_row = 0;
-    while (first_row < sel_row - num_rows + 1)
-        first_row += num_rows + 2;
+    while (first_row < sel_row - display_rows + 1)
+        first_row += display_rows + 1;
     // make sure the selection stays visible
     if (first_row > sel_row)
-        first_row = sel_row - num_rows + 1;
+        first_row = sel_row - display_rows + 1;
     // don't scroll beyond the end
-    if (first_row > units.size() - num_rows)
-        first_row = units.size() - num_rows;
+    if (first_row > units.size() - display_rows)
+        first_row = units.size() - display_rows;
 
     last_selection = -1;
+
+    read_manipulator_config();
+}
+
+void viewscreen_unitlaborsst::calcArrivals()
+{
+    int bi=-100;
```

Comment:
What do these names mean?

Reply:
I could comment them and/or call them single letters, but they are simply defined by their use in the code which very shortly follows their declaration here. They are bai :before active index, cai: current active index, bi before index, ci: current index